### ISSUE TITLE
Segmentation fault when trying to parse custom dates

### ISSUE THREAD

**Issue by acelaya** [CuyZ/Valinor#234]
I'm trying to make valinor map `DateTimeInterface` properties to [cakephp/chronos](https://github.com/cakephp/chronos) objects, by following [this documentation](https://valinor.cuyz.io/latest/other/dealing-with-dates/#custom-date-class-implementation), but when configure it to infer the type, it halts the script with a segmentation fault when trying to map objects.

It can be reproduced with valinor 0.15, using this script under PHP 8.1

```php
<?php

declare(strict_types=1);

use Cake\Chronos\Chronos;
use CuyZ\Valinor\Mapper\Source\Source;
use CuyZ\Valinor\Mapper\Tree\Message\MessageBuilder;
use CuyZ\Valinor\MapperBuilder;

require __DIR__ . '/../vendor/autoload.php';

class User {
    public function __construct(
        public readonly string $email,
        public readonly DateTimeInterface $dateOfBirth,
        /** @var string[] */
        public readonly array $favoriteTopics,
        public readonly ?string $domain = null,
    ) {
    }
}

$data = [
    'email' => 'john.doe@gmail.com',
    'dateOfBirth' => Chronos::now()->startOfDay()->subYears(20)->toAtomString(),
    'favoriteTopics' => ['foo', 'bar'],
];

$mapper = (new MapperBuilder())
    ->infer(DateTimeInterface::class, static fn () => Chronos::class)
    ->registerConstructor(static fn ($date): Chronos => Chronos::parse($date))
    ->filterExceptions(static fn (Throwable $e) => MessageBuilder::from($e))
    ->mapper();

try {
    var_dump($mapper->map(User::class, Source::array($data)));
} catch (Throwable $e) {
    var_dump($e);
}
```

If the `infer`, `registerConstructor` and `filterExceptions` lines are commented out, then it works by mapping the date as a DateTimeImmutable object.

**Comment by romm** [comment]
Nice catch!

Should be fixed with #237 — any chance you can test it out on your side and confirm if it's ok?

**Comment by acelaya** [comment]
I sure can

**Comment by acelaya** [comment]
Yep, I can confirm it fixes it :)

**Comment by romm** [comment]
Thank you for the confirmation!

Now released as [0.16.0](https://github.com/CuyZ/Valinor/releases/tag/0.16.0). 😉
